django/forms/widgets.py

```diff
@@ -798,8 +798,12 @@ class MultiWidget(Widget):
     """
     template_name = 'django/forms/widgets/multiwidget.html'
 
-    def __init__(self, widgets, attrs=None):
+    def __init__(self, widgets, attrs=None, names=None):
```

Comment:
Any thoughts about allowing `widgets` to be a `dict` mapping names to widgets now that we assume that `dict`s are ordered?

That seems more elegant than introducing a `names` kwarg and _zip_'ing it with `widgets`.

Reply:
I've updated this to allow `widgets` to be a `dict` as you suggested. I've also considered backwards compatibility so where `widgets` is not a `dict` the old behaviour remains.